### ISSUE TITLE
Use a Secret in DockerHub Webhooks for Caller validation. 

### ISSUE THREAD

**Issue by mghazizadeh** [docker/roadmap#51]
We would like to use Webhooks to provide ready to deploy functions for DockerHub workflow automation for the community, but we need Hub Webhooks to use some kind of security encryption using a secret so that the caller (ie: DockeHub) can be validated. 
Open to any of these: 
Bearer Token
HAMC-SH1
HMAC-SH256
Content Type: application/json
Here is how GitHub does it: 
https://developer.github.com/webhooks/creating/?Secret#secret

see also this issue: 
https://github.com/docker/hub-feedback/issues/1161

**Comment by digvan** [comment]
+1

**Comment by ingshtrom** [comment]
@mghazizadeh Please follow the [issue template](https://github.com/docker/roadmap/blob/master/.github/ISSUE_TEMPLATE/feature_request.md) for creating new issues here. It helps us to parse through information quicker if all of our issues are formatted in a similar way.

@digvan please use reactions on the original issue comment. This helps us to track community response to various issues without cluttering an issue with lots of `+1` comments.

**Comment by mghazizadeh** [comment]
@ingshtrom done! modified to use the issue template.

**Comment by pkennedyr** [comment]
@mghazizadeh Thanks for submitting this issue - we'll investigate with the team!

**Comment by digvan** [comment]
@pkennedyr 
The way GitHub is securing the webhook is a good starting point. You can check this link for more information.
https://developer.github.com/webhooks/securing/

Here is also Stripe implementation (which is more secure). 
https://stripe.com/docs/webhooks/signatures

**Comment by Soumil07** [comment]
Any update on this?

**Comment by adwaitabhandare** [comment]
+1, 
Any update on this ? 
Github/Gitlab/Bitbucket all provide support for webhook secret auth.

**Comment by MarcoMatarazzo** [comment]
Pretty please?

**Comment by david-szabo97** [comment]
+1

**Comment by agustinaliagac** [comment]
+1

**Comment by codebysandip** [comment]
Sorry I'm saying this but it's 3 years and still no response on this important feature.
What I can say just remove this feature of webhook because it's useless until we are not able to verify a request coming from docker hub and someone just want my server to go down by sending multiple requests

**Comment by Kasperror** [comment]
How can such important company forget about such important feature for so many years. When buying DockerHub I thought ci/cd automation will be a piece of cake but no sane techie will allow unauthorized calls to run deploy automation. A workaround through GitHub's workflows after image push is a best option by now but WHY?

### Solution ideas:
1. Host endpoint with current IP ranges, like GitHub and many other do. 
2. Add secret key to webhooks (even an auto-generated one if y'all too busy)

I think @nebuk89 is the right person to actually take on implementing this feature.

**Comment by DanielinR** [comment]
I want this feature too :cake:  :c. Something basic would be enought for me.

**Comment by krab-skunk** [comment]
I cannot believe i'm here asking about this feature, we are not talking about some random flask server dude over internet here, we are talking about freaking Dockerhub, and basically not a single mechanism to ensure webhooks come from them,  worst not even some IP address to lock security groups on??? Seriously, WTF!!!

**Comment by atomix-dev** [comment]
no info till now then

**Comment by dmars8047** [comment]
Yeah I can't help but feel a little let down by Docker here. Kinda seems like the webhook feature is a little useless without some kind of HMAC security. Otherwise these webhook events could be from anyone.

**Comment by dmars8047** [comment]
Really need this feature. I can't create automation without it.

**Comment by fuskovic** [comment]
+1

**Comment by philong17** [comment]
+1
